torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -433,6 +443,10 @@ IndexCompute IndexCompute::updateIndexCompute(
 
     if (extent_map_.find(prev_id) != extent_map_.end()) {
       updated_extent_map[new_id] = extent_map_.at(prev_id);
+    } else {
+      if (prev_id->isReduction() && !new_id->isReduction()) {
+        updated_extent_map[new_id] = getExtent(prev_id);
+      }
```

Comment:
Is this sufficient to fix #456?

Reply:
I believe so, but that could simply be because I couldn't think of a counter example. This should introduce any dimensions anywhere in the iter domain maps associated with a reduction. If it is a reduction that should be the first point to introduce it as it wouldn't have been previously mapped. This also prevents it from being introduced when propagating from producer -> consumer where it wouldn't have to be introduced.